samples/retained-instance-store/src/main/java/com/badoo/ribs/samples/retained_instance_store/rib/retainedInstance/RetainedInstancePresenter.kt

```diff
@@ -0,0 +1,43 @@
+package com.badoo.ribs.samples.retained_instance_store.rib.retainedInstance
+
+import android.os.Parcelable
+import androidx.lifecycle.Lifecycle
+import com.badoo.ribs.android.subscribe
+import com.badoo.ribs.core.plugin.ViewAware
+import com.badoo.ribs.routing.source.backstack.BackStack
+import com.badoo.ribs.routing.source.backstack.operation.replace
+import com.badoo.ribs.samples.retained_instance_store.rib.retainedInstance.routing.RetainedInstanceRouter
+import java.lang.IllegalStateException
+
+interface RetainedInstancePresenter {
+    fun toggleChildCounter()
+}
+
+internal class RetainedInstancePresenterImpl(
+        private val backStack: BackStack<RetainedInstanceRouter.Configuration>
+) : ViewAware<RetainedInstanceView>,
+        RetainedInstancePresenter {
+
+    private var view: RetainedInstanceView? = null
+
+    override fun onViewCreated(view: RetainedInstanceView, viewLifecycle: Lifecycle) {
+        super.onViewCreated(view, viewLifecycle)
+        viewLifecycle.subscribe(
+                onCreate = { this.view = view },
+                onDestroy = { this.view = null }
+        )
+    }
+
+    override fun toggleChildCounter() {
+        val currentConfig = backStack.getCurrentConfig()
+        val newConfig = if (currentConfig is RetainedInstanceRouter.Configuration.Retained) {
+            RetainedInstanceRouter.Configuration.NotRetained
+        } else {
+            RetainedInstanceRouter.Configuration.Retained
```

Comment:
Let's import them as `Content.Retained` and `Content.NotRetained` (see my other comment in router)